code/modules/defenses/sentry_computer.dm

```diff
@@ -0,0 +1,420 @@
+
+/obj/item/device/sentry_computer/proc/teardown()
+	setup = FALSE
+	open = FALSE
+	on = FALSE
+	icon_state = "sentrycomp_cl"
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/device/sentry_computer/MouseDrop(atom/dropping, mob/user)
+	teardown()
+	usr.put_in_any_hand_if_possible(src, disable_warning = TRUE)
+
+/obj/item/device/sentry_computer/proc/handle_engaged(var/obj/structure/machinery/defenses/sentry/sentrygun)
+	var/displayname = sentrygun.name
+	if(length(sentrygun.nickname) > 0)
+		displayname = sentrygun.nickname
+	var/areaname = get_area(sentrygun)
+	var/message = "[displayname]:[areaname] Engaged [sentrygun.target]."
+	INVOKE_ASYNC(src, .proc/send_message, message)
+
+/obj/item/device/sentry_computer/proc/handle_low_ammo(var/obj/structure/machinery/defenses/sentry/sentrygun)
+	if(sentrygun.ammo)
+		var/displayname = sentrygun.name
+		if(length(sentrygun.nickname) > 0)
+			displayname = sentrygun.nickname
+		var/areaname = get_area(sentrygun)
+		var/message = "[displayname]:[areaname] Low ammo [sentrygun.ammo.current_rounds]/[sentrygun.ammo.max_rounds]."
+		INVOKE_ASYNC(src, .proc/send_message, message)
+
+/obj/item/device/sentry_computer/proc/handle_empty_ammo(var/obj/structure/machinery/defenses/sentry/sentrygun)
+	var/displayname = sentrygun.name
+	if(length(sentrygun.nickname) > 0)
+		displayname = sentrygun.nickname
+	var/areaname = get_area(sentrygun)
+	var/message = "[displayname]:[areaname] out of ammo."
+	INVOKE_ASYNC(src, .proc/send_message, message)
+
+/obj/item/device/sentry_computer/proc/send_message(var/message)
+	if(transceiver)
+		transceiver.talk_into(voice, "[message]", RADIO_CHANNEL_SENTY)
+
+/obj/item/device/sentry_computer/attack_hand(mob/user)
+	if(setup)
+		if(!on)
+			icon_state = "sentrycomp_on"
+			on = TRUE
+			START_PROCESSING(SSobj, src)
+		else
+			tgui_interact(user)
+	else
+		..()
+
+/obj/item/device/sentry_computer/get_examine_text()
+	. = ..()
+	if(cell)
+		. += "A [cell.name] is loaded. It has [cell.charge]/[cell.maxcharge] charge remaining."
+	else
+		. += "It has no battery inserted."
+
+/obj/item/device/sentry_computer/process()
+	if(on)
+		if(cell)
+			var/energy_cost = length(paired_sentry) * power_consumption * CELLRATE
+			if(cell.charge >= (energy_cost))
+				cell.use(energy_cost)
+			else
+				icon_state = "sentrycomp_op"
+				on = FALSE
+
+/obj/item/device/sentry_computer/attackby(var/obj/item/object, mob/user)
+	if(istype(object, /obj/item/cell))
+		var/obj/item/cell/new_cell = object
+		to_chat(user, SPAN_NOTICE("The new cell contains: [new_cell.charge] power."))
+		cell.forceMove(get_turf(user))
+		cell = new_cell
+		user.drop_inv_item_to_loc(new_cell, src)
+		playsound(src,'sound/machines/click.ogg', 25, 1)
+	else if(istype(object, /obj/item/device/multitool))
+		var/obj/item/device/multitool/tool = object
+		var/id = tool.serial_number
+		playsound(src, get_sfx("terminal_type"), 25, FALSE)
+		if (do_after(usr, 2, INTERRUPT_NO_NEEDHAND, BUSY_ICON_GENERIC))
```

Comment:
0.2 SECONDS

any value below 0.6 seconds does not trigger a do_after I believe (need to check this)

Reply:
I've set all these to 10, I didn't realise this was in deciseconds